shark_turbine/kernel/wave/expansion.py

```diff
@@ -208,14 +214,40 @@ def expand_graph(
                 dim: val for dim, val in zip(dim_scaling.keys(), dim_combination)
             }
             logger.debug(f"Starting expansion at leaf:{node} in dims:{expand_dims}")
-            _expand_node(node, trace, expand_dims, dim_scaling, expansion_context)
+            _expand_node(
+                node, trace, expand_dims, dim_scaling, dim_tile_size, expansion_context
+            )
+
+    # Replace symbolic tile sizes with actual tile sizes.
+    def replace_index(node: fx.Node) -> bool:
+        if node.index is not None:
+            for dim in node.index:
+                node.index[dim].start = node.index[dim].start.subs(tile_size_mapping)
+        return False
+
+    trace.walk(replace_index)
+
+
+def update_index(
```

Comment:
No action for now (More of a wish list really), Any thoughts on somehow combining `update_index` and `compute_index` in a single location/function?

Reply:
The reason why they are separate is because I wanted to avoid redundant computation. We can put compute_index into update_index but that means that we would have to repeat the constraint based index computation for every node in the expansion. By separating them out, we pay a one time cost to compute the constraint based indices and then only compute the expansion based indices during expansion.

Reply:
I see, yeah that makes sense, would've been nice for reading ergonomics though haha. 

Although, I think if we made the expand nodes less pinned on dimension-to-dimension, by trying to expand all the dims in one go and just move from node to node, then we may be able to combine both index compute and still save compute here. Although, I think it's a bit hard to convey what I am saying over message, we can chat more about this on Monday with some example IRs I was thinking about. :)

Reply:
This feels like a setter for the index attribute of CustomOp. Could we add this as a property there?
e.g.
```Python
    @property
    def index(self):
        return self._index

    @index.setter
    def index(self, new_index):
        // [...]
        self._index = ...
```

Reply:
Sure I can add it there.

Reply:
Style NIT on that, maybe we should name the index setter function as `set_index` for better clarity.

Reply:
@raikonenfnu - I believe that's how you define the setter (you can't do it as set_index https://realpython.com/python-getter-setter/)